DSNP-Wallet/Source/Common/ExtrinsicManager/ExtrinsicManager+Msa.swift

```
@@ -0,0 +1,61 @@
+//
+//  ExtrinsicManager+Msa.swift
+//  DSNP-Wallet
+//
+//  Created by Ryan Sheh on 2/22/23.
+//
+
+import Foundation
+import RobinHood
+import SubstrateSdk
+
+extension ExtrinsicManager {
+    func createMSA(completion completionClosure: @escaping ExtrinsicSubmitClosure) throws {
+        let closure: ExtrinsicBuilderClosure = { builder in
+            let call = self.callFactory.createMsa()
+            _ = try builder.adding(call: call)
+            return builder
+        }
+        
+        guard let signer = user?.signer else { throw ExtrinsicError.BadSetup }
+        
+        extrinsicService?.submit(closure,
+                                 signer: signer,
+                                 runningIn: .main,
+                                 completion: completionClosure)
+    }
+    
+    func addPublicKeyToMsa(primaryUser: UserFacadeProtocol, secondaryUser: UserFacadeProtocol, completion completionClosure: @escaping ExtrinsicSubmitClosure) throws {
+        guard let primarySigner = primaryUser.signer,
+              let secondarySigner = secondaryUser.signer,
+              let primaryPublicKeyData = primaryUser.publicKey?.rawData(),
+              let secondaryAccountId = secondaryUser.getAccountId() else {
+            throw ExtrinsicError.BadSetup
+        }
+        
+        let addKeyPayload = AddKeyPayloadArg(msaId: 1,
+                                             expiration: 5,
+                                             newPublicKey: secondaryAccountId)
+        
+        guard let keyPayloadData = try? JSONEncoder().encode(addKeyPayload) else { return }
```

perhaps we should throw an error if the encoding fails